[Bug]: for Python 3.11.9 gor error  ValueError: PyCapsule_New called with null pointer

### Bug summary

I got next error 
src\tests\test_cover.py:48: 
    _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
    src\tests\test_cover.py:202: in draw_diagram
        fig, ax = plt.subplots(figsize=(14, 5), layout='constrained')
    .venv\Lib\site-packages\matplotlib\pyplot.py:1702: in subplots
        fig = figure(**fig_kw)
    .venv\Lib\site-packages\matplotlib\pyplot.py:1022: in figure
        manager = new_figure_manager(
    .venv\Lib\site-packages\matplotlib\pyplot.py:545: in new_figure_manager
        return _get_backend_mod().new_figure_manager(*args, **kwargs)
    .venv\Lib\site-packages\matplotlib\backend_bases.py:3521: in new_figure_manager
        return cls.new_figure_manager_given_figure(num, fig)
    .venv\Lib\site-packages\matplotlib\backend_bases.py:3526: in new_figure_manager_given_figure
        return cls.FigureCanvas.new_manager(figure, num)
    .venv\Lib\site-packages\matplotlib\backend_bases.py:1811: in new_manager
        return cls.manager_class.create_with_canvas(cls, figure, num)
    .venv\Lib\site-packages\matplotlib\backends\_backend_tk.py:479: in create_with_canvas
        with _restore_foreground_window_at_end():
    C:\Users\olapbuild\AppData\Local\Programs\Python\Python311\Lib\contextlib.py:137: in __enter__
        return next(self.gen)
    _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
    
        @contextmanager
        def _restore_foreground_window_at_end():
    >       foreground = _c_internal_utils.Win32_GetForegroundWindow()
    E       ValueError: PyCapsule_New called with null pointer
    
    .venv\Lib\site-packages\matplotlib\backends\_backend_tk.py:43: ValueError

### Code for reproduction

```Python
def draw_diagram():
    width = 0.4  # Задаём ширину столбцов
    fig, ax = plt.subplots(figsize=(14, 5), layout='constrained')
    ax.bar(labels, cowered_data, width, label='Процент покрытых мер-измерений тестами')
    ax.bar(labels, uncowered_data, width, bottom=cowered_data,
           label='Процент непокрытых мер-измерений тестами')
    ax.set_ylabel('% покрытия')
    ax.set_title('Процент покрытия кубов тестами')
    ax.legend()
    plt.savefig('diagram.png')
```


### Actual outcome

src\tests\test_cover.py:48: 
    _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
    src\tests\test_cover.py:202: in draw_diagram
        fig, ax = plt.subplots(figsize=(14, 5), layout='constrained')
    .venv\Lib\site-packages\matplotlib\pyplot.py:1702: in subplots
        fig = figure(**fig_kw)
    .venv\Lib\site-packages\matplotlib\pyplot.py:1022: in figure
        manager = new_figure_manager(
    .venv\Lib\site-packages\matplotlib\pyplot.py:545: in new_figure_manager
        return _get_backend_mod().new_figure_manager(*args, **kwargs)
    .venv\Lib\site-packages\matplotlib\backend_bases.py:3521: in new_figure_manager
        return cls.new_figure_manager_given_figure(num, fig)
    .venv\Lib\site-packages\matplotlib\backend_bases.py:3526: in new_figure_manager_given_figure
        return cls.FigureCanvas.new_manager(figure, num)
    .venv\Lib\site-packages\matplotlib\backend_bases.py:1811: in new_manager
        return cls.manager_class.create_with_canvas(cls, figure, num)
    .venv\Lib\site-packages\matplotlib\backends\_backend_tk.py:479: in create_with_canvas
        with _restore_foreground_window_at_end():
    C:\Users\olapbuild\AppData\Local\Programs\Python\Python311\Lib\contextlib.py:137: in __enter__
        return next(self.gen)
    _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
    
        @contextmanager
        def _restore_foreground_window_at_end():
    >       foreground = _c_internal_utils.Win32_GetForegroundWindow()
    E       ValueError: PyCapsule_New called with null pointer
    
    .venv\Lib\site-packages\matplotlib\backends\_backend_tk.py:43: ValueError

### Expected outcome

![diagram](https://github.com/matplotlib/matplotlib/assets/105438413/cca8af94-9e66-441f-9c5a-8790bdb67ffc)


### Additional information

_No response_

### Operating system

Windows NT

### Matplotlib Version

3.9.0

### Matplotlib Backend

tkagg

### Python version

3.11.9, 3.12

### Jupyter version

_No response_

### Installation

pip

--- a/lib/matplotlib/backends/_backend_tk.py
+++ b/lib/matplotlib/backends/_backend_tk.py
@@ -44,7 +44,7 @@ def _restore_foreground_window_at_end():
     try:
         yield
     finally:
-        if mpl.rcParams['tk.window_focus']:
+        if foreground and mpl.rcParams['tk.window_focus']:
             _c_internal_utils.Win32_SetForegroundWindow(foreground)
 
 

--- a/src/_c_internal_utils.cpp
+++ b/src/_c_internal_utils.cpp
@@ -111,7 +111,11 @@ static py::object
 mpl_GetForegroundWindow(void)
 {
 #ifdef _WIN32
-  return py::capsule(GetForegroundWindow(), "HWND");
+  if (HWND hwnd = GetForegroundWindow()) {
+    return py::capsule(hwnd, "HWND");
+  } else {
+    return py::none();
+  }
 #else
   return py::none();
 #endif

